Move damped harmonic rollon initialization action into separate file

nilsdeppe opened the pull request:
## Proposed changes

- The action that initializes the gauge should be in the same file that the gauge is in. This also helps de-clutter the `Initialize.hpp` file.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsdeppe committed:
Move damped harmonic rollon init action into separate file

geoffrey4444 commented:
clang-tidy failure

moxcodes commented on src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:
```diff
@@ -0,0 +1,111 @@
+  const auto lapse = gr::lapse(shift, spacetime_metric);
+  const auto inverse_spacetime_metric =
+      gr::inverse_spacetime_metric(lapse, shift, inverse_spatial_metric);
+  tnsr::abb<DataVector, Dim, Frame::Inertial> da_spacetime_metric{};
```
initialize with the size.

moxcodes commented on src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:
```diff
@@ -0,0 +1,111 @@
+    const Mesh<Dim>& mesh,
+    const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
+        inverse_jacobian) noexcept {
+  const auto spatial_metric = gr::spatial_metric(spacetime_metric);
```
There are several temporaries being allocated in a row in this function, though I believe this is typically only run once during the evolution, so perhaps the allocation cost is not too high.

moxcodes reviewed (dismissed):
Looks good

nilsdeppe committed:
Use Variables in setting initial gauge condition

nilsdeppe commented:
@moxcodes @geoffrey4444 thanks for the reviews! I've fixed the clang-tidy error and added a commit that uses a Variables when initializing the gauge. I think having that be a separate commit makes the move easier.

nilsdeppe commented:
@geoffrey4444 sorry I had to fix another clang-tidy error that I missed previously

moxcodes reviewed (dismissed):
Great!

kidder commented:
@nilsdeppe needs a rebase

nilsdeppe commented:
Okay, rebased. Thanks again for the reviews :)
